internal/rule/exists.go

```diff
@@ -0,0 +1,144 @@
+package rule
+
+import (
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+type Exists struct {
+	name      string
+	exclusive bool
+	min       int16
+	max       int16
+	count     int16
+	*sync.RWMutex
+}
+
+func (rule *Exists) Init() Rule {
+	rule.name = "exists"
+	rule.exclusive = true
+	rule.count = 0
+	rule.RWMutex = new(sync.RWMutex)
+
+	return rule
+}
+
+func (rule *Exists) GetName() string {
+	rule.RLock()
+	defer rule.RUnlock()
+
+	return rule.name
+}
+
+// 0 = regex pattern
+func (rule *Exists) SetParameters(params []string) error {
+	rule.Lock()
+	defer rule.Unlock()
+
+	// exists
+	if len(params) == 0 {
+		rule.min = 1
+		rule.max = math.MaxInt16
+		return nil
+	}
+
+	// exists:
+	if params[0] == "" {
+		return fmt.Errorf("exists value is empty")
+	}
+
+	// exists:1
+	var split = strings.Split(params[0], "-")
+	if len(split) == 1 {
+		var value int64
+		var err error
+
+		if value, err = strconv.ParseInt(params[0], 10, 16); err != nil {
+			return err.(*strconv.NumError).Err
```

Comment:
You should return err

And use errors.As to catch them

Or you use a type assertion check


```suggestion
     if nerr, ok := err.(*strconv.NumError); ok {
			    return nerr.Err
     }
     return err
```


I'm a bit worried by the possible panic here if strconv slightly changes